public/app/plugins/panel/canvas/editor/options.ts

```diff
@@ -1,19 +1,31 @@
 import { PanelOptionsSupplier } from '@grafana/data/src/panel/PanelPlugin';
-import { BackgroundImageSize, CanvasElementOptions } from 'app/features/canvas';
+import { CanvasElementOptions } from 'app/features/canvas';
 import { ColorDimensionEditor, ResourceDimensionEditor } from 'app/features/dimensions/editors';
 
+import { BackgroundSizeEditor } from '../../../../features/dimensions/editors/BackgroundSizeEditor';
+
 interface OptionSuppliers {
   addBackground: PanelOptionsSupplier<CanvasElementOptions>;
   addBorder: PanelOptionsSupplier<CanvasElementOptions>;
 }
 
+const getCategoryName = (str: string, type: string | undefined) => {
+  if (type !== 'frame' && type !== undefined) {
+    return [str + ` (${type})`];
+  }
+
+  return [str];
+};
+
+const uid = Math.floor(Math.random() * 21);
+
 export const optionBuilder: OptionSuppliers = {
   addBackground: (builder, context) => {
-    const category = ['Background'];
+    const category = getCategoryName('Background', context.options?.type);
```

Comment:
Is this necessary change? I think we can keep the category the same and rely on context instead (for what the options apply to)

Reply:
It looks like as long as category is unique the functionality works as expected

With that in mind I removed the uid approach, though we should probably have another person test this locally to confirm these findings (before merge) - I'm going to approve this though as is 👍